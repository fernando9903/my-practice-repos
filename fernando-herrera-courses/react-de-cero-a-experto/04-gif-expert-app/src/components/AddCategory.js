import React, { useState } from "react";

export const AddCategory = () => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const handleOnSubmit = (event) => {
        event.preventDefault();
        console.log('Submit hecho');
        setInputValue('');
    }

    return (
        <form onSubmit={handleOnSubmit}>
            <input
                type='text'
                placeholder='Type a new category that you want to add and press enter'
                onChange={handleInputChange}
            />
        </form>
    );
}