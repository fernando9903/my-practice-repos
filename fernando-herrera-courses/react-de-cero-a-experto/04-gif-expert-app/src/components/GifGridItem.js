import React from "react";
import PropTypes from 'prop-types';

export const GridItem = ({title, url}) => {

    return (
        <div className='card'>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}