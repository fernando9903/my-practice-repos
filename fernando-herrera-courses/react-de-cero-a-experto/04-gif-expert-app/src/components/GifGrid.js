import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";

import { GridItem } from "./GifGridItem";

export const GifGrid = ({category}) => {

    const {loading, data:images} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {loading ? 'Loading...' :
            <div className='card-grid animate__animated animate__fadeInTopLeft'>
                {images.map( image => 
                    <GridItem
                        key={image.id}
                        {...image}
                    />
                )}
            </div>}
        </>
    )
}