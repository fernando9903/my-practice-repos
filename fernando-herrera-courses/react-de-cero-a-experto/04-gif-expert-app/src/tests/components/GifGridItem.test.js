import React from 'react';
import {shallow} from 'enzyme';
import { GridItem } from '../../components/GifGridItem';

describe('Test into <GifGridItem/> component', () => {
    const title = 'Some title';
    const url = 'https://localhost/some.jpg';

    const wraper = shallow(<GridItem 
        title={title}
        url={url}    
    />);
    
    test('The component should be render correclty', () => {
        expect(wraper).toMatchSnapshot();
    });

    test('The component should have the text into the p', () => {
        const p = wraper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('The img should have an src and alt equal to the test values', () => {
        const img = wraper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

});