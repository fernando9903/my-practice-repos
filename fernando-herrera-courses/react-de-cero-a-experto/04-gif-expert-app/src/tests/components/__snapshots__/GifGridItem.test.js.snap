// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test into <GifGridItem/> component The component should be render correclty 1`] = `
<div
  className="card"
>
  <img
    alt="Some title"
    src="https://localhost/some.jpg"
  />
  <p>
    Some title
  </p>
</div>
`;
